### PR TITLE
Prometheus: Annotation query filtering values less then 1

### LINKED ISSUE #43139: Prometheus Annotations that are based on a value between 0 and 1 do not show
**What happened**:
I created an annotation based on a Prometheus gauge metrics with values between 0 and 1.
image

The annotation didn't appear (I cross refrense it with a panel)
image

I then changed the annotation to be:
``10*min(scylla_node_ops_finished_percentage) by (ops, dc,instance) < 10``
And the annotation appear:
image

**What you expected to happen**:
``min(scylla_node_ops_finished_percentage) by (ops, dc,instance) < 1`` should have work
**How to reproduce it (as minimally and precisely as possible)**:
Create an annotation based on a prometheus value between 0 and 1.

**Anything else we need to know?**:

**Environment**:
- Grafana version: 8.3.2 (but it didn't work in other 8.x versions that I tried)
- Data source type & version: prometheus
- OS Grafana is installed on: docker
- User OS & Browser:
- Grafana plugins:
- Others:

### PR DESCRIPTION
**Pull request by gtk-grafana**
**What this PR does / why we need it**:
Annotations don't work when prometheus is returning values between zero and one.

Fixes https://github.com/grafana/grafana/issues/43139

**Special notes for your reviewer**:
What a doozy of a review, I apologize in advance for not splitting this out! :P

### CODE REVIEW

**Comment by gtk-grafana** [comment]
With fix:
image

Without fix:
image

Annotation 1:
image

Annotation 2:
image

File: public/app/plugins/datasource/prometheus/datasource.tsx [hunk 1]
```diff
@@ -797,7 +797,7 @@ export class PrometheusDatasource
         timeValueTuple.push([timeStampValue, valueValue]);
       });
 
-      const activeValues = timeValueTuple.filter((value) => value[1] >= 1);
+      const activeValues = timeValueTuple.filter((value) => value[1] >= 0);
```
Comment (gtk-grafana) [inline public/app/plugins/datasource/prometheus/datasource.tsx:800]: @ivanahuckova Is there some context as to why we were filtering out values less than 1?

**Comment by gtk-grafana** [comment]
Failing unit tests, pulling into draft :(

**Comment by gtk-grafana** [comment]
Fixed unit tests locally, marking ready for review

File: public/app/plugins/datasource/prometheus/datasource.tsx (hunk 1 above)
Comment (gtk-grafana) [reply public/app/plugins/datasource/prometheus/datasource.tsx]: Oops, I had a bug in here, we didn't want to include 0 values

**Review by obetomuniz — APPROVED** [review]
LGTM ✅

File: public/app/plugins/datasource/prometheus/datasource.tsx [hunk 2]
```diff
@@ -797,7 +797,7 @@ export class PrometheusDatasource
         timeValueTuple.push([timeStampValue, valueValue]);
       });
 
-      const activeValues = timeValueTuple.filter((value) => value[1] >= 1);
+      const activeValues = timeValueTuple.filter((value) => value[1] > 0);
```
Comment (gtk-grafana) [inline public/app/plugins/datasource/prometheus/datasource.tsx:800]: @ivanahuckova @torkelo Is there any historical context for why we are filtering values greater then 0, or should this be fixed?

Comment (torkelo) [reply public/app/plugins/datasource/prometheus/datasource.tsx:800]: no idea

Comment (ivanahuckova) [reply public/app/plugins/datasource/prometheus/datasource.tsx:800]: I did line history check and it seems that it was introduced in https://github.com/grafana/grafana/pull/22225 that fixed https://github.com/grafana/grafana/issues/18174.

Comment (gtk-grafana) [reply public/app/plugins/datasource/prometheus/datasource.tsx:800]: Thanks @ivanahuckova! It looks as though that issue was to prevent a bunch of zero values from being returned as they can slow things down. When testing similar queries on this branch vs main I'm not seeing any difference in behavior. 

@secustor it looks like you were the author of this change, can you think of any problems that might be introduced if we change this to allow values greater than 0?

Comment (secustor) [reply public/app/plugins/datasource/prometheus/datasource.tsx:800]: > @secustor it looks like you were the author of this change, can you think of any problems that might be introduced if we change this to allow values greater than 0?     

Nothing from top of my head. 

That said, the screenshots with fix look wrong. Is the step size 15s and datapoint interval 30s in the example? In that case logic would still be working correctly, else it should be a region annotation

Comment (gtk-grafana) [reply public/app/plugins/datasource/prometheus/datasource.tsx:800]: Yeah I might have messed up those screenshots, let me try again using a consistent step size:

Query:
image

Annotation 1:
image

Annotation 2:
image

I'm still not sure I'm fully understanding here, @secustor does this help clarify?

Comment (secustor) [reply public/app/plugins/datasource/prometheus/datasource.tsx:800]: LGTM

**Comment by gtk-grafana** [comment]
@grafana/alerting-squad-frontend Is this going to cause any trouble for y'all?

**Comment by gtk-grafana** [comment]
@grafana/alerting-squad any concerns with this?

**Comment by yuri-tceretian** [comment]
cc @grafana/alerting-squad-frontend

**Review by secustor — APPROVED** [review]

**Comment by gillesdemey** [comment]
Nope, this looks good – we don't do anything funky with Prometheus values :)

**Review by gillesdemey — APPROVED** [review]
